classes/Group.php

```diff
@@ -137,6 +137,10 @@ public static function getReductionByIdGroup($id_group)
 
     public static function getPriceDisplayMethod($id_group)
     {
+        // if the feature "Customer Groups" is disabled so we need to change the id_group to PS_CUSTOMER_GROUP **default behavior**
+        if (!Group::isFeatureActive()) {
+            $id_group = (int)Configuration::get('PS_CUSTOMER_GROUP');
```

Comment:
This method is meant  to return the display method for a given group. The code asking for it WANTS the display method for THIS given group. If the method internally changes `$id_group` and returns display method for this new group, then the code asking for if cannot trust the method anymore.
Determining the group id should be done before calling `getPriceDisplayMethod()`. First, you find the correct group id, and then you ask for its display method.